meta/meta.go

```diff
@@ -117,6 +117,23 @@ func (m *Meta) GenGlobalID() (int64, error) {
 	return m.txn.Inc(mNextGlobalIDKey, 1)
 }
 
+// GenGlobalIDs generates the next n global IDs.
+func (m *Meta) GenGlobalIDs(n int) ([]int64, error) {
+	globalIDMutex.Lock()
+	defer globalIDMutex.Unlock()
+
+	new, err := m.txn.Inc(mNextGlobalIDKey, int64(n))
+	if err != nil {
+		return nil, err
+	}
+	orig := new - int64(n)
```

Comment:
If there are several tidb instance genGlobalIDs simultaneously, could this condition hold?

Reply:
Since we are holding the lock, this should be OK. I also modify the test case to make the two `GenGlobalIDs` call concurrently.